analysis/quantifier-comparisons/pre-processing-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,612 @@
+  geom_boxplot() + 
+  theme_classic() + 
+  ylab("Number of Genes detected/Cell") + 
+  xlab("")
+```
+
+
+```{r}
+ggplot(cell_qc_filter, aes(y = detected, fill = tool)) + 
+  geom_boxplot() + 
+  facet_wrap(~sample) + 
+  theme_classic() + 
+  ylab("Genes Detected/Cell") +
+  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
+  
+```
+
+So we see a lot more variation in the genes detected/cell across tools than we did in the UMI/cell. Alevin is highest in the same 126 and 127 samples and cellranger consistently seems to be on the low end. Alevin-fry in `--sketch` mode again seems to get more coverage than alevin-fry without `--sketch` mode. 
+
+Previously Josh had found that Kallisto detects more UMI's and more genes/cell but that doesn't always seem to be the case. It looks like in 126 and 127, Alevin and Alevin-fry `--sketch` are higher. But again, cellranger is keeping more cells than the other tools in those 2 samples. 
+
+What if we just look at cells that are found in all of the tools? 
+
+```{r}
+cell_counts <- cell_qc_filter %>%  
+  dplyr::count(cell_id, sample)
+
+common_cells <- cell_counts %>%
+  dplyr::filter(n == 5) %>%
+  dplyr::pull(cell_id)
+
+cell_qc_common <- cell_qc_filter %>%
+  dplyr::filter(
+    (cell_id %in% common_cells) 
+  )
+```
+
+Now what happens to our mito content, UMI/cell, and genes/cell? 
+
+```{r}
+# mito comparison across shared cells only of all 5 runs
+ggplot(cell_qc_common, aes(y = subsets_mito_percent, fill = tool)) + 
+  geom_boxplot() + 
+  facet_wrap(~sample) + 
+  theme_classic() + 
+  ylab("% Mito /Cell") + 
+  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
+```
+```{r}
+# number of UMI's/sample in shared cells only 
+ggplot(cell_qc_common, aes(x = tool, y = sum, fill = tool)) + 
+  geom_boxplot() + 
+  facet_wrap(~sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("UMI/cell") + 
+  xlab("")
+```
+```{r}
+# genes detected/cell in shared cells only 
+ggplot(cell_qc_common, aes(y = detected, fill = tool)) + 
+  geom_boxplot() + 
+  facet_wrap(~sample) + 
+  theme_classic() + 
+  ylab("Genes Detected/Cell") +
+  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
+  
+```
+Wow, so everything is very consistent when you look at only shared cells except for the genes/cell. Now, we actually see that alevin-fry `--sketch` captures the highest number of genes/cell in shared cells only, with cellranger, alevin, and alevin-fry without `--sketch` all capturing around the same. 
+
+Now that we've looked at the stats for the shared cells, let's look and see how many cells are shared across which tools.  
+
+```{r}
+# make a table with information about each cell identified by a tool and how many other tools 
+# that cell is identified in
+cell_counts <- cell_qc_filter %>%
+  dplyr::select(tool, cell_id, sum, detected, sample) %>%
+  left_join(cell_counts)
+```
+
+```{r}
+cell_counts <- cell_counts %>%
```

Comment:
I think grouping the samples here is hiding information. As with the other plots, keeping the samples separate at this stage seems the way to go. We know that alevin is detecting many fewer cells for whatever reason in some samples, so that may be skewing the plot for other samples.